From 45a4345c66423d8590da33e4aa14edc26ddaa283 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Thu, 29 Oct 2020 23:47:30 +0100
Subject: [PATCH] hostnamed: init chassis before guessing icon name

We check in guess_icon_name for the configured chassis so we must set
the chassis value first.
---
 src/hostnamed.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/hostnamed.c b/src/hostnamed.c
index cde34ce..878bc42 100644
--- a/src/hostnamed.c
+++ b/src/hostnamed.c
@@ -680,30 +680,30 @@ hostnamed_init (gboolean _read_only)
         err = NULL;
     }
 
-    icon_name = shell_source_var (machine_info_file, "${ICON_NAME}", &err);
-    if (icon_name == NULL)
-        icon_name = g_strdup ("");
+    chassis = shell_source_var (machine_info_file, "${CHASSIS}", &err);
+    if (chassis == NULL)
+        chassis = g_strdup ("");
     if (err != NULL) {
         g_debug ("%s", err->message);
         g_error_free (err);
         err = NULL;
     }
-    if (icon_name == NULL || *icon_name == 0) {
-        g_free (icon_name);
-        icon_name = guess_icon_name ();
+    if (chassis == NULL || *chassis == 0) {
+        g_free (chassis);
+        chassis = guess_chassis ();
     }
 
-    chassis = shell_source_var (machine_info_file, "${CHASSIS}", &err);
-    if (chassis == NULL)
-        chassis = g_strdup ("");
+    icon_name = shell_source_var (machine_info_file, "${ICON_NAME}", &err);
+    if (icon_name == NULL)
+        icon_name = g_strdup ("");
     if (err != NULL) {
         g_debug ("%s", err->message);
         g_error_free (err);
         err = NULL;
     }
-    if (chassis == NULL || *chassis == 0) {
-        g_free (chassis);
-        chassis = guess_chassis ();
+    if (icon_name == NULL || *icon_name == 0) {
+        g_free (icon_name);
+        icon_name = guess_icon_name ();
     }
 
     deployment = shell_source_var (machine_info_file, "${DEPLOYMENT}", &err);
-- 
2.25.1

